Src/PChecker/CheckerCore/SystematicTesting/Strategies/Feedback/Generator/ControlledRandom.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using PChecker.Generator.Mutator;
+using PChecker.Random;
+
+namespace PChecker.Generator.Object;
+
+/// <summary>
+/// Controlled Random.
+/// </summary>
+public class ControlledRandom: IRandomValueGenerator
+{
+    public RandomChoices<int> IntChoices;
+    public RandomChoices<double> DoubleChoices;
+    
+    /// <summary>
+    /// Constructor
+    /// </summary>
+    /// <param name="int"></param>
+    /// <param name="double"></param>
+    public ControlledRandom(RandomChoices<int> @int, RandomChoices<double> @double)
+    {
+        IntChoices = @int;
+        DoubleChoices = @double;
+    }
+
+    public ControlledRandom(CheckerConfiguration checkerConfiguration) : this(new System.Random((int?)checkerConfiguration.RandomGeneratorSeed ?? Guid.NewGuid().GetHashCode()))
+    {
+    }
+
+    public ControlledRandom(System.Random random)
+    {
+        IntChoices = new RandomChoices<int>(random);
+        DoubleChoices = new RandomChoices<double>(random);
+    }
+    
+    /// <inheritdoc/>
+    public uint Seed { get; set; }
```

Comment:
When is this `Seed` being set?

Reply:
Seed is not used in ControlledRandom. This field is added to implement the `IRandomValueGenerator` interface